packages/ui/src/overview/components/DeadlineList/ElectionListItem.tsx

```diff
@@ -48,39 +57,50 @@ export const ElectionListItem: React.FC<ElectionListItemProps> = React.memo(({ e
     }
   }, [])
 
-  const copyMessage = () => {
-    const urlAddress = '#/election?candidate='
+  const getCopyMessage = useMemo(() => {
+    const urlAddress = '#/election'
     switch (electionStage) {
       case 'announcing':
+        if (election.candidates.some((candidate) => candidate.member.id === member.id)) {
+          return null
+        }
         return (
           <StyledText>
-            {t('deadline.announcingPeriod', { electionName: title })}
-            <StyledLink href={urlAddress + electionId}>{t('deadline.announcingLink')}</StyledLink>
+            {t('deadline.announcingPeriod')}
+            <StyledLink href={urlAddress}>{t('deadline.announcingLink')}</StyledLink>
           </StyledText>
         )
       case 'voting':
+        if (!canVote) {
+          return null
+        }
         return (
           <StyledText>
-            {t('deadline.votingPeriod', { electionName: title })}{' '}
-            <StyledLink href={urlAddress + electionId}>{t('deadline.votingLink')}</StyledLink>
+            {t('deadline.votingPeriod')} <StyledLink href={urlAddress}>{t('deadline.votingLink')}</StyledLink>
           </StyledText>
         )
       case 'revealing':
+        if (votes?.every((vote) => vote.voteFor)) {
```

Comment:
That's a small detail but there shouldn't be a `?` here because if `votes` was `undefined` (which it can't be) this logic would be wrong
```suggestion
        if (votes.every((vote) => vote.voteFor)) {
```

Reply:
I think this hook ```const { votes } = useMyCastVotes(electionStage === 'revealing' ? election.cycleId : undefined) ?? []``` could return undefined value